event/oracle/upgrade_oracle_event.go

```diff
@@ -0,0 +1,94 @@
+package oracle
+
+import (
+	"errors"
+	oracletypes "github.com/medibloc/panacea-core/v2/x/oracle/types"
+	"github.com/medibloc/panacea-doracle/event"
+	"github.com/medibloc/panacea-doracle/panacea"
+	log "github.com/sirupsen/logrus"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+)
+
+type UpgradeOracleEvent struct {
+	service reactor
+}
+
+var _ event.Event = (*UpgradeOracleEvent)(nil)
+
+func NewUpgradeOracleEvent(s reactor) UpgradeOracleEvent {
+	return UpgradeOracleEvent{s}
+}
+
+func (e UpgradeOracleEvent) GetEventType() string {
+	return "message"
+}
+
+func (e UpgradeOracleEvent) GetEventAttributeKey() string {
+	return "action"
+}
+
+func (e UpgradeOracleEvent) GetEventAttributeValue() string {
+	return "'UpgradeOracle'"
+}
+
+func (e UpgradeOracleEvent) EventHandler(event ctypes.ResultEvent) error {
+	addressValue := event.Events[oracletypes.EventTypeUpgradeVote+"."+oracletypes.AttributeKeyOracleAddress][0]
+	service := e.service
+	queryClient := service.QueryClient()
+
+	uniqueID := service.EnclaveInfo().UniqueIDHex()
+	oracleRegistration, err := queryClient.GetOracleRegistration(addressValue, uniqueID)
+	if err != nil {
+		return err
+	}
+
+	voteOption, err := e.verifyAndGetVoteOption(oracleRegistration)
+	if err != nil {
+		return err
+	}
+
+	msgVoteOracleRegistration, err := makeOracleRegistrationVote(
+		uniqueID,
+		e.service.OracleAcc().GetAddress(),
+		addressValue,
+		voteOption,
+		e.service.OraclePrivKey().Serialize(),
+		oracleRegistration.NodePubKey,
+		oracleRegistration.Nonce,
+	)
+	if err != nil {
+		return err
+	}
+
+	txBuilder := panacea.NewTxBuilder(*e.service.QueryClient())
+
+	txBytes, err := generateTxBytes(msgVoteOracleRegistration, e.service.OracleAcc().GetPrivKey(), e.service.Config(), txBuilder)
+	if err != nil {
+		return err
+	}
+
+	if err := broadcastTx(e.service.GRPCClient(), txBytes); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (e UpgradeOracleEvent) verifyAndGetVoteOption(r *oracletypes.OracleRegistration) (oracletypes.VoteOption, error) {
+	upgradeInfo, err := e.service.QueryClient().GetOracleUpgradeInfo()
+	if err != nil {
+		if errors.Is(err, panacea.ErrEmptyValue) {
+			log.Infof("not exist oracle upgrade info")
+			return oracletypes.VOTE_OPTION_NO, nil
+		}
+		log.Errorf("failed to get oracle upgrade info. %v", err)
+		return oracletypes.VOTE_OPTION_NO, nil
```

Comment:
If the error is `panacea.ErrEmptyValue`, I think this should return `NO` as is.
But in case of other error, how about returning `oracletypes.VOTE_OPTION_UNSPECIFIED`? or do not vote because there is an unexpected error while getting oracle upgrade info.

Reply:
Well on the blockchain there is no handling of `oracletypes.VOTE_OPTION_UNSPECIFIED`. 
Once this vote is received, we cannot tally the votes.
As @H4NLee said, it seems right not to vote if an unexpected error occurs.

Reply:
Good 👍. Let's just log the unexpected error then.